### PR TITLE
[ALLUXIO-2023] Renaming and refactoring of authorization classes.

### PR DESCRIPTION
**Pull request by jsimsa**
Addresses: https://alluxio.atlassian.net/browse/ALLUXIO-2023

This gist of this PR is that it does the following renaming:

- PermissionStatus becomes Permission
- FileSystemPermission becomes Mode, renaming the permission field of PermissionStatus (now Permission) to mode
- FileSystemAction becomes Bits, nesting this class under Mode

### CODE REVIEW

File: core/common/src/main/java/alluxio/Constants.java [hunk 1]
```diff
@@ -459,18 +459,18 @@
...
       "alluxio.security.authorization.permission.umask";
   public static final String SECURITY_AUTHORIZATION_PERMISSION_SUPERGROUP =
       "alluxio.security.authorization.permission.supergroup";
   // Group Mapping
   public static final String SECURITY_GROUP_MAPPING = "alluxio.security.group.mapping.class";
 
   // Security related constant value
-  public static final int DEFAULT_FS_PERMISSIONS_UMASK = 0022;
-  public static final short DEFAULT_FS_FULL_PERMISSION = (short) 0777;
+  public static final int DEFAULT_FS_UMASK = 0022;
```
Comment (apc999) [inline core/common/src/main/java/alluxio/Constants.java]: NIT: FS -> FILESYSTEM?

Comment (jsimsa) [inline core/common/src/main/java/alluxio/Constants.java]: done

**Comment by ifcharming** [comment]
Thanks @jsimsa for the refactoring. The new names LGTM, made code much cleaner.

File: core/common/src/main/java/alluxio/Constants.java (hunk 1 above)
Comment (apc999) [inline core/common/src/main/java/alluxio/Constants.java]: is this PR up to date? I don't see this changed

File: core/common/src/main/java/alluxio/security/authorization/Mode.java [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+    private static final Bits[] SVALS = values();
+
+    Bits(String s) {
+      mString = s;
+    }
+
+    /**
+     * @return the string representation of the bits
+     */
+    public String toString() {
```
Comment (apc999) [inline core/common/src/main/java/alluxio/security/authorization/Mode.java]: `@Override`?

File: core/common/src/main/java/alluxio/Constants.java (hunk 1 above)
Comment (jsimsa) [inline core/common/src/main/java/alluxio/Constants.java]: forgot to `git push`, should be changed now

File: core/common/src/main/java/alluxio/security/authorization/Mode.java (hunk 2 above)
Comment (jsimsa) [inline core/common/src/main/java/alluxio/security/authorization/Mode.java]: done

**Comment by apc999** [comment]
LGTM

**Pull request merged**